templates/default/openstack-cloud-init.erb

```diff
@@ -7,6 +7,8 @@ ssh_pwauth: false
 locale_configfile: /etc/sysconfig/i18n
 mount_default_fields: [~, ~, 'auto', 'defaults,nofail', '0', '2']
 
+manage_etc_hosts: true
```

Comment:
Are we sure that we want to regenerate it on reboots?  Perhaps setting this to [localhost](http://bazaar.launchpad.net/~cloud-init-dev/cloud-init/trunk/view/head:/doc/examples/cloud-config.txt#L470) is a better option?